pygmt/accessors.py

```diff
@@ -29,6 +31,8 @@ def __init__(self, xarray_obj):
         self._obj = xarray_obj
         try:
             self._source = self._obj.encoding["source"]  # filepath to NetCDF source
+            if not Path(self._source).exists():
+                raise ValueError(f"Grid source file {self._source} doesn't exist.")
```

Comment:
Suggest to use https://docs.python.org/3/library/exceptions.html#FileNotFoundError instead. Also need a unit test for this line to ensure code coverage. 
```suggestion
            if not Path(self._source).exists():
                raise FileNotFoundError(f"Grid source file {self._source} doesn't exist.")
```